### PR TITLE
test limits.

### PR DESCRIPTION
**Pull request by gandrewstone**
First test checks generating and submitting a very large block.

### CODE REVIEW

**Comment by ptschip** [comment]
@gandrewstone I think we should add limits.py to the extended test suite in rpc-tests.py.

File: qa/rpc-tests/limits.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    t = MyTest()
+    bitcoinConf = {
+        "debug": ["net", "blk", "thin", "mempool", "req", "bench", "evict"],
+        "blockprioritysize": 2000000  # we don't want any transactions rejected due to insufficient fees...
+    }
+
+    # you may want these additional flags:
+    # "--srcdir=<out-of-source-build-dir>/debug/src"
+    # "--tmpdir=/ramdisk/test", "--nocleanup", "--noshutdown"
+    t.main(["--tmpdir=/ramdisk/test", "--srcdir=/fast/bitcoin/bugas/debug/src"], bitcoinConf, None)
```
Comment (sickpig) [inline qa/rpc-tests/limits.py]: these paths seems to be specific to your dev environment.

File: qa/rpc-tests/limits.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+# Copyright (c) 2015-2017 The Bitcoin Unlimited developers
```
Comment (sickpig) [inline qa/rpc-tests/limits.py]: 2018

File: qa/rpc-tests/limits.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+from test_framework.blocktools import *
+
+class MyTest (BitcoinTestFramework):
+
+    def setup_chain(self,bitcoinConfDict=None, wallets=None):
+        print("Initializing test directory "+self.options.tmpdir)
+        initialize_chain_clean(self.options.tmpdir, 4, bitcoinConfDict, wallets)
+        # Number of nodes to initialize ----------> ^
```
Comment (sickpig) [inline qa/rpc-tests/limits.py]: really like the way this comment looks :+1:

File: qa/rpc-tests/limits.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        while size < txBytes:
+            count += 1
+            utxo = wallet.pop()
+            outp = {}
+            # Make the tx bigger by adding addtl outputs so it validates faster
+            payamt = satoshi_round(utxo["amount"] / decimal.Decimal(8.0))
+            for x in range(0, 8):
+                # its test code, I don't care if rounding error is folded into the fee
+                outp[addrs[(count + x) % len(addrs)]] = payamt
+                #outscript = self.wastefulOutput(addrs[(count+x)%len(addrs)])
```
Comment (sickpig) [inline qa/rpc-tests/limits.py]: why not removing the following 3 commented lines of code

File: qa/rpc-tests/limits.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+                #outscripthex = hexlify(outscript).decode("ascii")
+                #outp[outscripthex] = payamt
+            if data:
+                outp["data"] = data
+            txn = createrawtransaction([utxo], outp, createWastefulOutput)
+            # txn2 = node.createrawtransaction([utxo], outp)
```
Comment (sickpig) [inline qa/rpc-tests/limits.py]: same for the line above ^^^

File: qa/rpc-tests/limits.py [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+    def makeUtxos(self, node, addrs, NUM_BLOCKS):
+
+        outp = {}
+        for a in addrs:
+            outp[a] = 0.1
+
+        for x in range(0,NUM_BLOCKS):
+            tx = node.sendmany("", outp)
+            # print(tx)
```
Comment (sickpig) [inline qa/rpc-tests/limits.py]: logging.info ?

File: qa/rpc-tests/limits.py [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+        n1blockcount = self.nodes[1].getblockcount()
+        ret = self.nodes[1].submitblock(bigBlockHex)
+        assert ret != "duplicate"
+        assert_equal(n1blockcount + 1, self.nodes[1].getblockcount())
+        assert_equal(self.nodes[0].getblockcount(), self.nodes[1].getblockcount())
+        return True
+
+
+if __name__ == '__main__':
+    MyTest ().main ()
```
Comment (sickpig) [inline qa/rpc-tests/limits.py]: nit: sometime a space is placed between function name and brackets sometime is not

Comment (gandrewstone) [reply qa/rpc-tests/limits.py]: good catch.  I will autopep8 the file

**Pull request merged**